packages/vite/src/node/server/hmr.ts

```diff
@@ -467,18 +466,19 @@ async function readModifiedFile(file: string): Promise<string> {
 async function restartServer(server: ViteDevServer) {
   // @ts-ignore
   global.__vite_start_time = Date.now()
+  
+  await server.close()
```

Comment:
@tjk just to check, I think the current pattern was used to avoid missing requests from the client, no? Did you analyze what happened in this case? The user may get a few network errors but the page is reloaded?